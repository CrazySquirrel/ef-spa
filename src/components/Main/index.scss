@import "../../scss/base";

.main {
  order: 2;
  padding: 1em 1em 0 1em;
  overflow-x: hidden;
  overflow-y: auto;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  @include display-flex;
  @include flex-direction(column);
}
