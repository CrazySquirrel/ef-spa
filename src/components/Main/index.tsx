import * as React from 'react';
import * as bem from 'bem-cn';

import './index.scss';

export default class Main extends React.Component {
  public render() {
    const block = bem('main');

    return (
        <main className={block()}>
          {this.props.children}
        </main>
    );
  }
}
