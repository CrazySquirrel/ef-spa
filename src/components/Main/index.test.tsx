import * as React from 'react';
import {shallow} from 'enzyme';

import Main from './index';

describe('Main', () => {
  it('render', () => {
    const result = shallow(<Main/>);

    expect(result).toMatchSnapshot();
  });
});
