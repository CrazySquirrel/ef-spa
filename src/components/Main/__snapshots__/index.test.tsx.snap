version https://git-lfs.github.com/spec/v1
oid sha256:f7fbe4476a38295d9bdc6e59a81faa652825ca1030693f16baf630fa716176fd
size 1436
